pilot/cmd/pilot-agent/status/server.go

```diff
@@ -527,6 +528,23 @@ func (s *Server) handleQuit(w http.ResponseWriter, r *http.Request) {
 	notifyExit()
 }
 
+func (s *Server) handleTcpProbe(w http.ResponseWriter, req *http.Request) {
+	portStr := strings.TrimPrefix(req.URL.Path, "/tcp-probe/")
+
+	port, err := strconv.Atoi(portStr)
+	if err != nil {
+		log.Errorf("could not parse port number when handling %s", req.URL.Path)
+	}
+
+	conn, err := net.Dial("tcp", fmt.Sprintf("localhost:%d", port))
```

Comment:
We need to use `s.appProbersDestination`, not localhost. https://istio.io/latest/blog/2021/upcoming-networking-changes/ for a long explanation about why that is

Reply:
Thanks